three-cards-divination/src/main/java/com/cyber/mysticism/tarot/controller/ThreeCardsDivinationController.java

```diff
@@ -20,7 +26,28 @@ public ThreeCardsDivinationController(ThreeCardsDivinationService threeCardsDivi
     }
 
     @GetMapping("/three-cards-divination")
-    public Map<String, Card> performReading() throws DivinationException {
-        return threeCardsDivinationService.getReading();
+    public Map<String, Card> performReadingForUser(@RequestHeader(value = "Authorization", required = false) String authHeader)
+            throws DivinationException, ForbiddenAccessException, UnauthorisedAccessException {
+        if (authHeader == null) {
+            throw new ForbiddenAccessException("Access forbidden!");
+        } else if (!authHeader.isEmpty() && authHeader.toLowerCase().startsWith("basic")) {
+            String base64Credentials = authHeader.substring("Basic".length()).trim();
+            String[] credentials = getCredentialsFromHeaders(base64Credentials);
+            String username = credentials[0];
+            String email = credentials[1];
+            try {
+                return threeCardsDivinationService.getReadingForUser(username, email);
+            } catch (UserNotFoundException e) {
+                throw new UnauthorisedAccessException("Unauthorized!");
+            }
+        } else {
+            throw new UnauthorisedAccessException("Unauthorized!");
+        }
```

Comment:
**Suggestion:** The `performReadingForUser` method in `ThreeCardsDivinationController` has a high cyclomatic complexity due to multiple if-else conditions. Consider refactoring this method to reduce complexity and improve readability. One way to do this is by extracting the logic for validating and processing the `authHeader` into a separate method. [maintainability]
```suggestion
    @GetMapping("/three-cards-divination")
    public Map<String, Card> performReadingForUser(@RequestHeader(value = "Authorization", required = false) String authHeader)
            throws DivinationException, ForbiddenAccessException, UnauthorisedAccessException {
        validateAuthHeader(authHeader);
        String base64Credentials = authHeader.substring("Basic".length()).trim();
        String[] credentials = getCredentialsFromHeaders(base64Credentials);
        String username = credentials[0];
        String email = credentials[1];
        return getReadingForUser(username, email);
    }

    private void validateAuthHeader(String authHeader) throws ForbiddenAccessException, UnauthorisedAccessException {
        if (authHeader == null) {
            throw new ForbiddenAccessException("Access forbidden!");
        } else if (authHeader.isEmpty() || !authHeader.toLowerCase().startsWith("basic")) {
            throw new UnauthorisedAccessException("Unauthorized!");
        }
    }

    private Map<String, Card> getReadingForUser(String username, String email) throws UnauthorisedAccessException {
        try {
            return threeCardsDivinationService.getReadingForUser(username, email);
        } catch (UserNotFoundException e) {
            throw new UnauthorisedAccessException("Unauthorized!");
        }
    }
```